### PR TITLE
bandwidth_limit: make exception free

### PR DESCRIPTION
**Pull request by mathetake**
Commit Message: bandwidth_limit: make exception free
Additional Description:
Risk Level: low
Testing: done
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Comment by mathetake** [comment]
@adisuissa explicit initialization done in https://github.com/envoyproxy/envoy/pull/37666/commits/d673aeba8be5fccd6519cd2ec04a5da2b03a2056 - agree it's better to do so than otherwise

File: source/extensions/filters/http/bandwidth_limit/bandwidth_limit.h [hunk 1]
```diff
@@ -65,7 +65,7 @@ class FilterConfig : public ::Envoy::Router::RouteSpecificFilterConfig {
 
   FilterConfig(const envoy::extensions::filters::http::bandwidth_limit::v3::BandwidthLimit& config,
```
Comment (alyssawilk) [inline source/extensions/filters/http/bandwidth_limit/bandwidth_limit.h:66]: @yanavlasov had a request that we not have required check-after-creation status checks in public constructors so after debate we've been making constructors private and having public static absl::StatusOr<std::unique_ptr<classname>> create() functions which use the protected constructor

also mild preference for creation status being the last entry just for consistency

/wait

Comment (mathetake) [reply source/extensions/filters/http/bandwidth_limit/bandwidth_limit.h]: done. Thank you for the review!

File: source/extensions/filters/http/bandwidth_limit/bandwidth_limit.cc [hunk 2]
```diff
@@ -57,7 +58,8 @@ FilterConfig::FilterConfig(const BandwidthLimit& config, Stats::Scope& scope,
                                                    DefaultResponseFilterDelayTrailer.get()))),
       enable_response_trailers_(config.enable_response_trailers()) {
   if (per_route && !config.has_limit_kbps()) {
-    throw EnvoyException("bandwidthlimitfilter: limit must be set for per route filter config");
+    creation_status = absl::InvalidArgumentError("limit must be set for per route filter config");
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply source/extensions/filters/http/bandwidth_limit/bandwidth_limit.cc:73]: not necessary given the rewrite but no objection to leaving it in

**Pull request merged**